sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+  AZ_IOT_HUB_CLIENT_STATUS_SERVER_ERROR = 500,
+  AZ_IOT_HUB_CLIENT_STATUS_BAD_GATEWAY = 502,
+  AZ_IOT_HUB_CLIENT_STATUS_SERVICE_UNAVAILABLE = 503,
+  AZ_IOT_HUB_CLIENT_STATUS_TIMEOUT = 504,
+} az_iot_hub_client_status;
+
+/**
+ * @brief Checks if the status indicates a successful operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the status indicates success. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_hub_client_is_success_status(az_iot_hub_client_status status);
+
+/**
+ * @brief Checks if the status indicates a retriable error occurred during the operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the operation should be retried. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_hub_client_is_retriable_status(az_iot_hub_client_status status);
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to twin operation responses.
+ *
+ * @param client
+ * @param mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                          filter.
+ * @param out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_twin_response_subscribe_topic_filter_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to twin desired property changes.
+ * @note The payload will contain only changes made to the desired properties.
+ *
+ * @param client
+ * @param mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                          filter.
+ * @param out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_twin_patch_subscribe_topic_filter_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief Twin response type.
+ *
+ */
+typedef enum
+{
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET = 1,
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES = 2,
+  AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES = 3,
+} az_iot_hub_client_twin_response_type;
+
+/**
+ * @brief Twin response.
+ *
+ */
+typedef struct az_iot_hub_client_twin_response
+{
+  az_iot_hub_client_twin_response_type response_type; /** @brief Twin response type. */
+  az_iot_hub_client_status status; /** @brief The operation status. */
+  az_span request_id; /** @brief Request ID matches the ID specified when issuing a Get or Patch
+                         command. */
+  az_span version; /** @brief The Twin object version.
+                    * @note This is only returned when
+                    * response_type==AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES or
+                    * response_type==AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES */

```

Comment:
>AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES [](start = 37, length = 52)

I thought the $version was only stamped globally to the twin and on desired?  What is meaning of it for a reported?

Reply:
Ohh - or is this from mqtt topic itself, not the payload?  If so would be interested on naming.

---
In reply to: [387280421](https://github.com/Azure/azure-sdk-for-c/pull/409#discussion_r387280421) [](ancestors = 387280421)

Reply:
Yes: this is from topic, _not_ payload. I think it's actually called "version" in the topic.